yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -257,6 +266,27 @@ export class Archiver implements ArchiveSource {
     }
   }
 
+  /** Checks if there'd be a reorg for the next block submission and start pruning now. */
+  private async handleEpochPrune(provenBlockNumber: bigint, currentL1BlockNumber: bigint) {
+    const localPendingBlockNumber = BigInt(await this.getBlockNumber());
+
+    const canPrune =
+      localPendingBlockNumber > provenBlockNumber &&
+      (await this.rollup.read.canPrune({ blockNumber: currentL1BlockNumber }));
```

Comment:
You will be running into the same issue as much of the other simulation, that the timestamp for this is in the past. 

Essentially, you figure out that you can prune but only after the fact. In reality, you already know that it *will* be pruned when the next block land, so you should probably do the same as those others. @just-mitch ran into a similar issue with the proving claims. #9193

Reply:
Filed https://github.com/AztecProtocol/aztec-packages/issues/9308